agent/installer/installer.go

```diff
@@ -103,29 +100,28 @@ func New(bundleRepo, downloadPath string, logger logr.Logger) (*installer, error
 		return nil, ErrDetectOs
 	}
 
-	return newUnchecked(os, bundleRepo, downloadPath, logger, &logPrinter{logger})
+	return newUnchecked(os, downloadPath, logger, &logPrinter{logger})
 }
 
 // newUnchecked returns an installer bypassing os detection and checks of bundleRepo and downloadPath.
 // If they are empty, returned installer will runs in preview mode, i.e.
 // executes everything except the actual commands.
-func newUnchecked(currentOs, bundleRepo, downloadPath string, logger logr.Logger, outputBuilder algo.OutputBuilder) (*installer, error) {
-	bd := bundleDownloader{repoAddr: bundleRepo, downloadPath: downloadPath, logger: logger}
-
-	reg := getSupportedRegistry(&bd, outputBuilder)
+func newUnchecked(currentOs, downloadPath string, logger logr.Logger, outputBuilder algo.OutputBuilder) (*installer, error) {
+	reg := getSupportedRegistry(outputBuilder)
 	if len(reg.ListK8s(currentOs)) == 0 {
 		return nil, ErrOsK8sNotSupported
 	}
 
 	return &installer{
-		algoRegistry:     reg,
-		bundleDownloader: bd,
-		detectedOs:       currentOs,
-		logger:           logger}, nil
+		algoRegistry: reg,
+		detectedOs:   currentOs,
+		logger:       logger}, nil
 }
 
 // Install installs the specified k8s version on the current OS
-func (i *installer) Install(k8sVer, tag string) error {
+func (i *installer) Install(bundleRepo, k8sVer, tag string) error {
+	i.bundleDownloader = bundleDownloader{repoAddr: bundleRepo, downloadPath: i.downloadPath, logger: i.logger}
```

Comment:
it's bad design that this method modifies the installer object. Same for uninstall.

Reply:
Agreed. This needs to be refactored. I think the old assumption was that we would have the `repoAddr` when the host-agent was started. But this assumption was false. The `repoAddr` will be passed in at Install / Uninstall time, so there is no point saving it. We just took the path of least resistance to get the installer integrated.

Reply:
Created #221 to track this work

Reply:
I acknowledge the least resistance for FC. That was to New every time. A much smaller and safer change.